tests/flytekit/unit/core/test_launch_plan.py

```diff
@@ -37,3 +36,62 @@ def wf(a: int) -> (str, str):
 
     lp_with_defaults = launch_plan.LaunchPlan.create("get_or_create2", wf, default_inputs={"a": 3})
     assert lp_with_defaults.parameters.parameters["a"].default.scalar.primitive.integer == 3
+
+
+def test_lp_all_parameters():
+    @task
+    def t1(a: int) -> typing.NamedTuple("OutputsBC", t1_int_output=int, c=str):
+        a = a + 2
+        return a, "world-" + str(a)
+
+    @task
+    def t2(a: str, b: str, c: str) -> str:
+        return b + a + c
+
+    @workflow
+    def wf(a: int, c: str) -> str:
+        x, y = t1(a=a)
+        u = t2(a=x, b=y, c=c)
+        return u
+
+    obj = CronSchedule("* * ? * * *", kickoff_time_input_arg="abc")
+    obj1 = CronSchedule("10 * ? * * *", kickoff_time_input_arg="abc")
+    slack_notif = notification.Slack(
+        phases=[_execution_model.WorkflowExecutionPhase.SUCCEEDED], recipients_email=["my-team@slack.com"]
+    )
+    auth_role_model = AuthRole(assumable_iam_role="my:iam:role")
+
+    lp = launch_plan.LaunchPlan.get_or_create(
+        workflow=wf,
+        name="get_or_create",
+        default_inputs={"a": 3},
+        fixed_inputs={"c": "4"},
+        schedule=obj,
+        notifications=slack_notif,
+        auth_role=auth_role_model,
+    )
+    lp2 = launch_plan.LaunchPlan.get_or_create(
+        workflow=wf,
+        name="get_or_create",
+        default_inputs={"a": 3},
+        fixed_inputs={"c": "4"},
+        schedule=obj,
+        notifications=slack_notif,
+        auth_role=auth_role_model,
+    )
+
+    assert lp is lp2
+
+    # Check for assertion error when a different scheduler is used
+    lp3 = launch_plan.LaunchPlan.get_or_create(
```

Comment:
can we add a test for each item separately?  Like two launch plans with `fixed_inputs` filled in, but all the other optional fields empty.  I'm just trying to make sure we capture any issues around Nones.

Reply:
Added `test_lp_each_parameter()` to handle the concerned cases.